src/test/tst_edge.cpp

```diff
@@ -43,8 +45,130 @@ void TestEdge::initTestCase()
     m_nodeB = std::make_shared<UNode>(1, Attributes());
 }
 
+void TestEdge::tst_edge()
```

Comment:
you could break your 3 test cases into three smaller methods like `tst_edge1()`, `tst_edge2()` and `tst_edge3()`